### PR TITLE
Most Drive Commands, Camera, and network table for vision

### PR DESCRIPTION
**Pull request by andrewmartin5**
Added listed drive commands, as well as a camera and network table, which can be found in robot.java

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -19,23 +27,38 @@
...
      */
     @Override
     public void robotInit() {
         // Initialize OI here
-
+        camera1 = CameraServer.getInstance().startAutomaticCapture(0);
+        camera2 = CameraServer.getInstance().startAutomaticCapture(1);
+        videoSink = CameraServer.getInstance().getServer();
+        NetworkTableInstance inst = NetworkTableInstance.getDefault();
+        NetworkTable table = inst.getTable("Table Thingy");
```
Comment (msoucy) [inline src/main/java/frc/robot/Robot.java]: What does this table do?

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -65,4 +88,21 @@ public void teleopInit() {
             autonomousCommand.cancel();
         }
     }
+
+    public Command doANetworkTableThing() {
```
Comment (msoucy) [inline src/main/java/frc/robot/Robot.java]: When is this command used?

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+    @Override
+    public void initDefaultCommand() {
+
+        // Set the default command for a subsystem here.
+        // setDefaultCommand(new MySpecialCommand());
+    }
+
+    public Command driveNormal() {
+        // The command is named "Sample Command" and requires this subsystem.
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: No it's not

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        return new Command("driveNormal", this) {
+            @Override
+            protected void initialize() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: If `initialize`, `execute`, or `end` is empty, just remove the function entirely

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+            @Override
+            protected void end() {
+                // Called once after isFinished returns true
+            }
+        };
+    }
+
+    public Command goXDistance(double distance) {
+        // The command is named "Sample Command" and requires this subsystem.
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: Comment

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+            @Override
+            protected void execute() {
+                drive.arcadeDrive(sandstormSpeed, gyroCorrection);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                // Make this return true when this Command no longer needs to run execute()
+                if (leftEncoder.get() + rightEncoder.get() / 2 > distance)
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: Remove comment, replace entire function with `return (leftEncoder.get() + rightEncoder.get()) / 2 > distance;`

Comment (msoucy) [reply src/main/java/frc/robot/subsystems/Drive.java]: You need parenthesis when getting the average

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+            @Override
+            protected void execute() {
+                drive.arcadeDrive(0, gyroCorrection);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                // Make this return true when this Command no longer needs to run execute()
+                if (gyro.getAngle() == degrees)
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: - Remove comment
- There's a function called `gyroDrivePID.onTarget()`, you should return that instead. That's the purpose of the setpoint

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: See line 161

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+
+    public Command retractPiston() {
+        return new InstantCommand("Retract Piston", this, () -> {
+            climbPiston.set(Value.kReverse);
+        });
+    }
+
+    public Command score() {
+        // The command is named "Sample Command" and requires this subsystem.
+        return new Command("Score", this) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java]: 1. Create command `score()`
2. Fill out functions
3. ???
4. Profit

What does this function do?

**Review by msoucy — APPROVED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/Robot.java]: Fancy stuffs for the cool vision code sean is writing

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 4 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 3 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 5 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 6 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 7 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 8 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/Robot.java]: Actually I really dont know please explain

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/Robot.java]: I have a better question: what does it do?

Comment (andrewmartin5) [reply src/main/java/frc/robot/Robot.java]: ·

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 9 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

File: src/main/java/frc/robot/maps/PracticeBot.java [hunk 10]
```diff
@@ -44,5 +54,46 @@ public DigitalInput getGamepieceLimitSwitch() {
...
+                return SendableSpeedController.wrap(new SpeedControllerGroup(new WPI_TalonSRX(1), new WPI_TalonSRX(2)));
+            }
+        
+            @Override
+            public Lidar getLidar() {
+                return new Lidar(Port.kOnboard, 0x10);
+            }
+        
+            @Override
+            public SendableSpeedController getLeft() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/PracticeBot.java:117]: Please move this next to getRight()?

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+        gyroCorrection = value;
+    });
+
+    double sandstormSpeed = .2;
+
+    @Override
+    public void initDefaultCommand() {
+
+        // Set the default command for a subsystem here.
+        // setDefaultCommand(new MySpecialCommand());
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Drive.java]: You should set `driveNormal` as the default command.

File: src/main/java/frc/robot/Robot.java [hunk 12]
```diff
@@ -17,25 +26,40 @@
...
+    public static XboxController driveController = new XboxController(1);
+
     private Command autonomousCommand;
     final private SendableChooser<Command> chooser = new SendableChooser<>();
 
+    UsbCamera Camera0;
+    UsbCamera Camera1;
+    VideoSink videoSink;
+    boolean activeCamera = false;
+    boolean camera0Active = true;
```
Comment (bot190) [inline src/main/java/frc/robot/Robot.java:42]: Are both of these used??

**Review by bot190 — COMMENTED** [review]
A few more comments.

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 11 above)
Comment (andrewmartin5) [reply src/main/java/frc/robot/subsystems/Drive.java]: ·

**Review by bot190 — APPROVED** [review]

**Comment by ghost** [comment]
Noticed that the drive X Distance command will only work in the forward direction... I would probably just change the name to include "forward" as we likely won't need to go backward. If we do, just make another command.

**Pull request merged**